### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by polomarcus**
Use docker compose to start the scrapper, the ingest to the DB, and streamlit :

### Why
Easier onboarding

### How to use it ?
```
docker compose up streamlit # start streamlit
docker compose up ingest_to_db # save data to dockerize PG (database)
docker compose up sitemap # collect data
```

image

### CODE REVIEW

File: postgres/insert_existing_data_example.py [hunk 1]
```diff
@@ -37,12 +43,14 @@ def insert_data_in_sitemap_table(df_to_insert: pd.DataFrame, password: str):
 
         # create INSERT INTO table (columns) VALUES('%s',...)
         insert_stmt = "INSERT INTO {} ({}) {}".format(table, columns, values)
+
+        #@TODO create a utils to connect to the DB
         conn = psycopg2.connect(
-            database="quotaclimat",
-            user="root",
+            database=DB_DATABASE,
```
Comment (estellerambier) [inline postgres/insert_existing_data_example.py:50]: 🙃

File: quotaclimat/data_ingestion/ingest_db/ingest_sitemap_in_db.py [hunk 2]
```diff
@@ -11,12 +13,17 @@
...
 args = parser.parse_args()
-DB_PWD = args.dbpwd
+
+DB_PWD = os.environ.get('POSTGRES_PASSWORD', args.dbpwd)
 
 
 def run():
     for media, sitemap_conf in SITEMAP_CONFIG.items():
         try:
+            # is DB init with schemas ?
```
Comment (estellerambier) [inline quotaclimat/data_ingestion/ingest_db/ingest_sitemap_in_db.py]: No. the schemas were there https://github.com/dataforgoodfr/quotaclimat/tree/main/postgres/schemas

File: quotaclimat/data_ingestion/scaleway_db_backups.py [hunk 3]
```diff
@@ -13,16 +13,20 @@
 parser.add_argument("-a", "--accesskey")
 
 args = parser.parse_args()
-DB_PWD = args.dbpwd
+DB_DATABASE = os.environ.get('POSTGRES_DB', "quotaclimat")
+DB_USER = os.environ.get('POSTGRES_USER', "root")
+DB_PWD = os.environ.get('POSTGRES_PASSWORD', args.dbpwd)
+DB_HOST = os.environ.get('POSTGRES_HOST', "212.47.253.253")
+DB_PORT = os.environ.get('POSTGRES_PORT', "49154")
```
Comment (estellerambier) [inline quotaclimat/data_ingestion/scaleway_db_backups.py:20]: There is a way to share secret within GitHub without exposing them. Do you know what I am referring to?

File: quotaclimat/data_processing/sitemap/sitemap_processing.py [hunk 4]
```diff
@@ -142,7 +142,7 @@ def _migrate_from_pd_to_json(df_all: pd.DataFrame):
         landing_path = "sitemap_dumps/media_type=%s" % row[1]["media_type"]
         if not os.path.exists(landing_path):
             os.makedirs(landing_path)
-        with open(landing_path + "/%s.json" % row[1]["media"], "w") as f:
+        with open(landing_path + "/%s.json" % row[1]["media"], "w+") as f:
```
Comment (estellerambier) [inline quotaclimat/data_processing/sitemap/sitemap_processing.py:145]: There are two jobs for ingesting data from the sitemap: one for the database, and another for GitHub. The latter was in place before we had the database, writing data to GitHub. I didn't removed the pieces of code properly, but the GitHub one shouldn't be kept

File: quotaclimat/data_ingestion/scaleway_db_backups.py (hunk 3 above)
Comment (polomarcus) [reply quotaclimat/data_ingestion/scaleway_db_backups.py:20]: Yes I guess https://github.com/polomarcus/barometre-medias-quota-climat/settings/secrets/actions

I was thinking using the CI/CD the same way `args.dbpwd` is used

File: quotaclimat/data_processing/sitemap/sitemap_processing.py (hunk 4 above)
Comment (polomarcus) [reply quotaclimat/data_processing/sitemap/sitemap_processing.py:145]: For now, I've just deleted the `.github/workflows/scrap_sitemap.yml` file

**Pull request merged**